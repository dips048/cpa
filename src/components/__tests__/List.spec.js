import React from "react"
import renderer from "react-test-renderer"
import List from "../list"
import TestRenderer from "react-test-renderer"
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
import Offcanvas from "react-bootstrap/Offcanvas"
import CallLink from "../callLink"
import { render, fireEvent } from "@testing-library/react"

describe("List", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<List />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test("should renders `FontAwesomeIcon` & `CallLink' components", () => {
    const testRenderer = TestRenderer.create(<List />)
    const testInstance = testRenderer.root
    expect(testInstance.findAllByType(FontAwesomeIcon).length).toEqual(5)
    expect(testInstance.findByType(CallLink)).toBeTruthy()
    expect(testInstance.findByType(Offcanvas)).toBeTruthy()
  })

  test("on clicking `Menu button` it should open off canvas", () => {
    const { getByText, getByTestId } = render(
      <List menuLinks={[]} comapanyName={"MyCompany"} />
    )
    const menuButton = getByTestId("fa-icon-menu")
    expect(menuButton).toBeInTheDocument()
    fireEvent.click(menuButton);
    expect(getByText("Menu")).toBeInTheDocument();
  })
})
