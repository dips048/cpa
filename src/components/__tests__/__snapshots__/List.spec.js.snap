// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly 1`] = `
<div>
  <div
    className="d-flex justify-content-between d-lg-none py-2"
  >
    <span
      className="flex-grow-1 text-center icon-color"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-lock fa-sm "
        data-icon="lock"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M80 192V144C80 64.47 144.5 0 224 0C303.5 0 368 64.47 368 144V192H384C419.3 192 448 220.7 448 256V448C448 483.3 419.3 512 384 512H64C28.65 512 0 483.3 0 448V256C0 220.7 28.65 192 64 192H80zM144 192H304V144C304 99.82 268.2 64 224 64C179.8 64 144 99.82 144 144V192z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
    <span
      className="flex-grow-1 text-center icon-color"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-rss fa-sm "
        data-icon="rss"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M25.57 176.1C12.41 175.4 .9117 185.2 .0523 198.4s9.173 24.65 22.39 25.5c120.1 7.875 225.7 112.7 233.6 233.6C256.9 470.3 267.4 480 279.1 480c.5313 0 1.062-.0313 1.594-.0625c13.22-.8438 23.25-12.28 22.39-25.5C294.6 310.3 169.7 185.4 25.57 176.1zM32 32C14.33 32 0 46.31 0 64s14.33 32 32 32c194.1 0 352 157.9 352 352c0 17.69 14.33 32 32 32s32-14.31 32-32C448 218.6 261.4 32 32 32zM63.1 351.9C28.63 351.9 0 380.6 0 416s28.63 64 63.1 64s64.08-28.62 64.08-64S99.37 351.9 63.1 351.9z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
    <a
      className="flex-grow-1 text-center icon-color"
      href="mailto:support@cpasitesolutions.com"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-envelope fa-sm "
        data-icon="envelope"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M464 64C490.5 64 512 85.49 512 112C512 127.1 504.9 141.3 492.8 150.4L275.2 313.6C263.8 322.1 248.2 322.1 236.8 313.6L19.2 150.4C7.113 141.3 0 127.1 0 112C0 85.49 21.49 64 48 64H464zM217.6 339.2C240.4 356.3 271.6 356.3 294.4 339.2L512 176V384C512 419.3 483.3 448 448 448H64C28.65 448 0 419.3 0 384V176L217.6 339.2z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
    <a
      className="flex-grow-1 text-center icon-color"
      href="tel:(800) 896-4500"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-phone fa-sm "
        data-icon="phone"
        data-prefix="fas"
        data-testid="fa-icon"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M511.2 387l-23.25 100.8c-3.266 14.25-15.79 24.22-30.46 24.22C205.2 512 0 306.8 0 54.5c0-14.66 9.969-27.2 24.22-30.45l100.8-23.25C139.7-2.602 154.7 5.018 160.8 18.92l46.52 108.5c5.438 12.78 1.77 27.67-8.98 36.45L144.5 207.1c33.98 69.22 90.26 125.5 159.5 159.5l44.08-53.8c8.688-10.78 23.69-14.51 36.47-8.975l108.5 46.51C506.1 357.2 514.6 372.4 511.2 387z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
    <span
      className="flex-grow-1 text-center icon-color"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-bars fa-sm "
        data-icon="bars"
        data-prefix="fas"
        data-testid="fa-icon-menu"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 96C0 78.33 14.33 64 32 64H416C433.7 64 448 78.33 448 96C448 113.7 433.7 128 416 128H32C14.33 128 0 113.7 0 96zM0 256C0 238.3 14.33 224 32 224H416C433.7 224 448 238.3 448 256C448 273.7 433.7 288 416 288H32C14.33 288 0 273.7 0 256zM416 448H32C14.33 448 0 433.7 0 416C0 398.3 14.33 384 32 384H416C433.7 384 448 398.3 448 416C448 433.7 433.7 448 416 448z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
  </div>
</div>
`;
